src/log/File.cc

```diff
@@ -112,21 +112,35 @@ void
 logfilePrintf(Logfile * lf, const char *fmt,...)
 {
     va_list args;
-    char buf[8192];
-    int s;
-
     va_start(args, fmt);
 
-    s = vsnprintf(buf, 8192, fmt, args);
-
-    if (s > 8192) {
-        s = 8192;
+    // bufsize will be large enough for most log lines
+    int bufsize = 16 * 1024;
+    char* buf = NULL;
+
+    while (true) {
+        buf = new char[bufsize];
+        if (!buf) {
+            break;
+        }
+
+        auto s = vsnprintf(buf, bufsize, fmt, args);
+        if (s < bufsize) {
+            logfileWrite(lf, buf, (size_t) s);
+            delete buf;
```

Comment:
Please do not leave dangling pointers, especially if code checks them for being nil, even if the (current) code does not actually dereference them.

```suggestion
            delete buf;
            buf = nullptr;
```

You will need to refactor other code to accomodate this change, but I hope that this and other changes will become unnecessary after the higher level problems are addressed.